packages/studio-be/src/studio/cloud/cloud-client.ts

```diff
@@ -0,0 +1,125 @@
+import axios from 'axios'
+import { Logger } from 'botpress/sdk'
+import { isCDMError } from 'common/errors'
+import FormData from 'form-data'
+import { constants } from 'http2'
+
+import _ from 'lodash'
+import qs from 'qs'
+import { Result, Ok, Err } from 'ts-results'
```

Comment:
It brings the concept of type-safe errors in our codebase. I think it will help the reliability of our codebase if clients of an API can know in advance all error types returned by a function. More here https://blog.logrocket.com/improve-error-handling-typescript-exhaustive-type-checking/